exopy_qcircuits/tasks/tasks/instr/alazar_tasks.py

```diff
@@ -163,10 +165,123 @@ def check(self, *args, **kwargs):
+        tracetimeA = self.format_string(self.tracetimeaftertrig, 10**-9, 1)
+        tracedurationA = self.format_string(self.traceduration, 10**-9, 1)
+        tracetimeB = self.format_string(self.tracetimeaftertrigB, 10**-9, 1)
+        tracedurationB = self.format_string(self.tracedurationB, 10**-9, 1)
+        demodFormFile = self.format_and_eval_string(self.demodFormFile)
+
+        NdemodA = len(durationA)
+        if 0 in durationA:
+            NdemodA = 0
+            timeA = []
+            durationA = []
+        NdemodB = len(durationB)
+        if 0 in durationB:
+            NdemodB = 0
+            timeB = []
+            durationB = []
+        NtraceA = len(tracedurationA)
+        if 0 in tracedurationA:
+            NtraceA = 0
+            tracetimeA = []
+            tracedurationA = []
+        NtraceB = len(tracedurationB)
+        if 0 in tracedurationB:
+            NtraceB = 0
+            tracetimeB = []
+            tracedurationB = []
+        if len(demodFormFile)== 0:
+            demodCosinus = 1;
+        else:
+            demodCosinus = 0;
+
+        startaftertrig = timeA + timeB + tracetimeA + tracetimeB
+        duration = durationA + durationB + tracedurationA + tracedurationB
+        timestepA = self.format_string(self.timestep, 10**-9, NdemodA)
+        timestepB = self.format_string(self.timestepB, 10**-9, NdemodB)
+        timestep = timestepA + timestepB
+        freqA = self.format_string(self.freq, 10**6, NdemodA)
+        freqB = self.format_string(self.freqB, 10**6, NdemodB)
+        freq = freqA + freqB      
+        samplesPerSec = 500000000.0         
+        
+        #Initialize the RAM quantities
+        taille_data = 0
+        RAM_1=0 #RAM coming from demodulations
+        RAM_2=0 #RAM coming from traces
+        list_RAM=[] #storage of the RAM estimations from each demodulation/trace
+        
+        #Calculate the RAM needed
+        samplesPerDemod_RAM = []
+        
+        #Calculate RAM_1 (in MB) coming from demodulations
+        for i in range(NdemodA + NdemodB):
+            samplesPerDemod_RAM.append(int(samplesPerSec * duration[i]) )
+
+            if timestep[i]:
+                RAM_1 += 15.265243530273438*(duration[i]/200e-9)*(recordsPerCapture/20000)
+                list_RAM.append(15.265243530273438*(duration[i]/200e-9)*(recordsPerCapture/20000))
+                #15.265243530273438 MB = deepgetsizeof(data(NdemodA=1,duration=200e-9,recordsPerCapture=20000),set())/(1024*1024)
+
+            elif not demodCosinus:
+                RAM_1 += 15.265243530273438*(duration[i]/200e-9)*(recordsPerCapture/20000)
+                list_RAM.append(15.265243530273438*(duration[i]/200e-9)*(recordsPerCapture/20000))
+                #15.265243530273438 MB = deepgetsizeof(data(NdemodA=1,duration=200e-9,recordsPerCapture=20000),set())/(1024*1024)
+                
+            else:
+                # Check wheter it is possible to cut each record in blocks of size equal
```

Comment:
Demodulation requires strictly more RAM if I remember correctly because all the data is first saved into the RAM and only demodulated and averaged at the end. I don't understand all the code below but it seems like it might be underestimating the RAM requirement.